valhalla/mjolnir/util.h

```diff
@@ -131,12 +139,76 @@ bool load_spatialite(sqlite3* db_handle);
  * unusable afterwards.  Set to false if you need to perform protobuf operations after building tiles.
  * @return Returns true if no errors occur, false if an error occurs.
  */
-bool build_tile_set(const boost::property_tree::ptree& config,
+bool build_tile_set(const ptree& config,
                     const std::vector<std::string>& input_files,
                     const BuildStage start_stage = BuildStage::kInitialize,
                     const BuildStage end_stage = BuildStage::kValidate,
                     const bool release_osmpbf_memory = true);
 
+// The tile manifest is a JSON-serializable index of tiles to be processed during the build stage of
+// valhalla_build_tiles'. It can be used to distribute shard keys when building tiles with
+// parallelized, distributed batch processing. For example, a workflow orchestrator can partition
+// and distribute this manifest to workers in a 'build' stage worker pool.
+//
+// This file written out during 'constructedges' and a prerequisite for the 'build' stage run by
+// valhalla_build_tiles.
+//
+// Example manifest :
+//
+// {
+//   "tiles": [
+//     {
+//       "node_index": 0,
+//       "graphid": {
+//         "value": 5970538,
+//         "id": 0,
+//         "tile_id": 746317,
+//         "level": 2
+//       }
+//     }
+//   ]
+// }
+struct TileManifest {
+
+  std::map<baldr::GraphId, size_t> tileset;
+
+  std::string ToString() const {
+    baldr::json::ArrayPtr array = baldr::json::array({});
+    for (const auto& tile : tileset) {
+      const baldr::json::Value& graphid = tile.first.json();
+      const baldr::json::MapPtr& item = baldr::json::map(
+          {{"graphid", graphid}, {"node_index", static_cast<uint64_t>(tile.second)}});
```

Comment:
in a future PR we should also store the node_count in an `std::pair<size_t,size_t>` from sort graph. this will give us an idea of which tiles are larger and which are smaller which is useful for distributed processing (load balancing)

Reply:
Noted!